app/views/courses/show.html.haml

```diff
@@ -13,10 +13,10 @@
           %p
             %b= semester
             \:
-          - instances.each do |instance|
+          - instances.sort_by { |instance| instance.section }.each do |instance|
             Section #{instance.section} taught by #{instance.professor.name}
             %br
-            = instance.meetings.first.schedule
+            = instance.meetings.try(:first).try(:schedule) || 'TBA'
```

Comment:
Ya, I like that.
You can improve line 3 with this: `return 'TBA' if meetings.blank?`